Parsers/ASimNetworkSession/Parsers/vimNetworkSessionFortinetFortiGate.yaml

```diff
@@ -0,0 +1,288 @@
+    , "89","OSPFIGP"
+    , "90","Sprite-RPC"
+    , "91","LARP"
+    , "92","MTP"
+    , "93","AX.25"
+    , "94","IPIP"
+    , "95","MICP (deprecated)"
+    , "96","SCC-SP"
+    , "97","ETHERIP"
+    , "98","ENCAP"
+    , "99",""
+    , "100","GMTP"
+    , "101","IFMP"
+    , "102","PNNI"
+    , "103","PIM"
+    , "104","ARIS"
+    , "105","SCPS"
+    , "106","QNX"
+    , "107","A/N"
+    , "108","IPComp"
+    , "109","SNP"
+    , "110","Compaq-Peer"
+    , "111","IPX-in-IP"
+    , "112","VRRP"
+    , "113","PGM"
+    , "114",""
+    , "115","L2TP"
+    , "116","DDX"
+    , "117","IATP"
+    , "118","STP"
+    , "119","SRP"
+    , "120","UTI"
+    , "121","SMP"
+    , "122","SM (deprecated)"
+    , "123","PTP"
+    , "124","ISIS over IPv4"
+    , "125","FIRE"
+    , "126","CRTP"
+    , "127","CRUDP"
+    , "128","SSCOPMCE"
+    , "129","IPLT"
+    , "130","SPS"
+    , "131","PIPE"
+    , "132","SCTP"
+    , "133","FC"
+    , "134","RSVP-E2E-IGNORE"
+    , "135","Mobility Header"
+    , "136","UDPLite"
+    , "137","MPLS-in-IP"
+    , "138","manet"
+    , "139","HIP"
+    , "140","Shim6"
+    , "141","WESP"
+    , "142","ROHC"
+    , "143","Ethernet"
+    , "253",""
+    , "254",""
+    , "255","Reserved"];
+  let src_or_any=set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+  let dst_or_any=set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+  let Parser=(starttime:datetime=datetime(null), endtime:datetime=datetime(null), srcipaddr_has_any_prefix:dynamic=dynamic([]), dstipaddr_has_any_prefix:dynamic=dynamic([]), ipaddr_has_any_prefix:dynamic=dynamic([]), dstportnumber:int=int(null), hostname_has_any:dynamic=dynamic([]), dvcaction:dynamic=dynamic([]), eventresult:string='*', disabled:bool=false){
+    CommonSecurityLog
+    | where not(disabled)
+    | where (isnull(starttime) or TimeGenerated>=starttime) and (isnull(endtime) or TimeGenerated<=endtime)
+    | where DeviceVendor == "Fortinet" and DeviceProduct startswith "FortiGate" and AdditionalExtensions has "cat=traffic"
+    | where DeviceAction != "dns"
+    | extend temp_SrcMatch=has_any_ipv4_prefix(SourceIP,src_or_any)
+      , temp_DstMatch=has_any_ipv4_prefix(DestinationIP,dst_or_any)
+    | extend ASimMatchingIpAddr=case(
+        array_length(src_or_any) == 0 and array_length(dst_or_any) == 0 ,"-",
+        temp_SrcMatch and temp_DstMatch, "Both",
+        temp_SrcMatch, "SrcIpAddr",
+        temp_DstMatch, "DstIpAddr",
+        "No match"
+    )
+    | where ASimMatchingIpAddr != "No match" 
+    | project-away temp_*
+    | project Activity,AdditionalExtensions,DestinationIP,DestinationPort,DeviceAction,DeviceInboundInterface,DeviceName,DeviceOutboundInterface,DeviceProduct,DeviceVersion,LogSeverity,Protocol,ReceivedBytes,SentBytes,SourceIP,SourcePort,TimeGenerated
+    | lookup EventLookup on DeviceAction 
+    | where  (array_length(hostname_has_any)==0) and (eventresult=="*" or (DvcAction=="Allow" and eventresult=="Success") or (eventresult=="Failure"))

```

Comment:
1. I think you lost the dstportnumber filter somewhere.
2. Still no filter on dvcaction (this one just uses DvcAction for the EventResult filter)

Reply:
Added both and the sample data